Mathlib/Topology/ContinuousOn.lean

```diff
@@ -1100,62 +1067,54 @@ theorem Embedding.map_nhdsWithin_eq {f : α → β} (hf : Embedding f) (s : Set
 
 theorem OpenEmbedding.map_nhdsWithin_preimage_eq {f : α → β} (hf : OpenEmbedding f) (s : Set β)
     (x : α) : map f (𝓝[f ⁻¹' s] x) = 𝓝[s] f x := by
-  rw [hf.to_embedding.map_nhds_within_eq, image_preimage_eq_inter_range]
+  rw [hf.toEmbedding.map_nhdsWithin_eq, image_preimage_eq_inter_range]
   apply nhdsWithin_eq_nhdsWithin (mem_range_self _) hf.open_range
   rw [inter_assoc, inter_self]
 #align open_embedding.map_nhds_within_preimage_eq OpenEmbedding.map_nhdsWithin_preimage_eq
 
 theorem continuousWithinAt_of_not_mem_closure {f : α → β} {s : Set α} {x : α} :
     x ∉ closure s → ContinuousWithinAt f s x := by
   intro hx
-  rw [mem_closure_iff_nhdsWithin_neBot, ne_bot_iff, Classical.not_not] at hx
+  rw [mem_closure_iff_nhdsWithin_neBot, neBot_iff, not_not] at hx
   rw [ContinuousWithinAt, hx]
   exact tendsto_bot
 #align continuous_within_at_of_not_mem_closure continuousWithinAt_of_not_mem_closure
 
-theorem ContinuousOn.piecewise' {s t : Set α} {f g : α → β} [∀ a, Decidable (a ∈ t)]
-    (hpf : ∀ a ∈ s ∩ frontier t, Tendsto f (𝓝[s ∩ t] a) (𝓝 (piecewise t f g a)))
-    (hpg : ∀ a ∈ s ∩ frontier t, Tendsto g (𝓝[s ∩ tᶜ] a) (𝓝 (piecewise t f g a)))
-    (hf : ContinuousOn f <| s ∩ t) (hg : ContinuousOn g <| s ∩ tᶜ) :
-    ContinuousOn (piecewise t f g) s := by
-  intro x hx
-  by_cases hx' : x ∈ frontier t
-  · exact (hpf x ⟨hx, hx'⟩).piecewise_nhds_within (hpg x ⟨hx, hx'⟩)
-  · rw [← inter_univ s, ← union_compl_self t, inter_union_distrib_left] at hx⊢
-    cases hx
-    · apply ContinuousWithinAt.union
-      ·
-        exact
-          (hf x hx).congr (fun y hy => piecewise_eq_of_mem _ _ _ hy.2)
-            (piecewise_eq_of_mem _ _ _ hx.2)
-      · have : x ∉ closure (tᶜ) := fun h => hx' ⟨subset_closure hx.2, by rwa [closure_compl] at h⟩
-        exact
-          continuousWithinAt_of_not_mem_closure fun h =>
-            this (closure_inter_subset_inter_closure _ _ h).2
-    · apply ContinuousWithinAt.union
-      · have : x ∉ closure t := fun h =>
-          hx' ⟨h, fun h' : x ∈ interior t => hx.2 (interior_subset h')⟩
-        exact
-          continuousWithinAt_of_not_mem_closure fun h =>
-            this (closure_inter_subset_inter_closure _ _ h).2
-      ·
-        exact
-          (hg x hx).congr (fun y hy => piecewise_eq_of_not_mem _ _ _ hy.2)
-            (piecewise_eq_of_not_mem _ _ _ hx.2)
-#align continuous_on.piecewise' ContinuousOn.piecewise'
-
 theorem ContinuousOn.if' {s : Set α} {p : α → Prop} {f g : α → β} [∀ a, Decidable (p a)]
```

Comment:
Lean failed to find a `Decidable` instance to apply the `piecewise'` version here, so I swapped the lemmas.